### PR TITLE
Redirect some doc pages the R Markdown book

### PR DESCRIPTION
**Pull request by yihui**
Also added links to PowerPoint format.

The redirected pages have been copied to the rmarkdown book, and revised there.

### CODE REVIEW

**Comment by yihui** [comment]
Synchronizing these web pages with the book is trickier than the case of vignettes for both an R package and its pkgdown website. The mapping is not strictly one to one (one page may have been split into multiple sections in the book, or multiple pages may have been merged into one section), and the header levels may have been adjusted in the book (e.g., `#` to `##`, or vice versa).

To make it slightly less jarring, we can open these external links in external windows. On the current website, we have already got some external links, and readers will be taken to the different context if they click these links.

**Comment by yihui** [comment]
Okay, I'll try my best.

File: _redirects [hunk 1]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
 https://rmarkdown.netlify.com/*  https://rmarkdown.rstudio.com/:splat  301!
 http://rmarkdown.netlify.com/*   https://rmarkdown.rstudio.com/:splat  301!
 /flexdashboard    /flexdashboard/   301!
+
+/r_notebooks.html  https://bookdown.org/yihui/rmarkdown/notebook.html  301!
```
Comment (yihui) [inline _redirects]: Section header levels changed in the book (n -> n + 1).

Added figure/table captions, numbers, and cross-references.

File: _redirects [hunk 2]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/html_document_format.html https://bookdown.org/yihui/rmarkdown/html-document.html  301!
```
Comment (yihui) [inline _redirects]: Levels of section headers.

Figure/table captions.

Citations.

File: _redirects [hunk 3]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/pdf_document_format.html  https://bookdown.org/yihui/rmarkdown/pdf-document.html  301!
```
Comment (yihui) [inline _redirects]: Levels of section headers.

Figure/table captions.

Cross-references of sections.

File: _redirects [hunk 4]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/word_document_format.html https://bookdown.org/yihui/rmarkdown/word-document.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

File: _redirects [hunk 5]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/odt_document_format.html  https://bookdown.org/yihui/rmarkdown/opendocument-text-document.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

File: _redirects [hunk 6]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/rtf_document_format.html  https://bookdown.org/yihui/rmarkdown/rich-text-format-document.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

File: _redirects [hunk 7]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/markdown_document_format.html https://bookdown.org/yihui/rmarkdown/markdown-document.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

Figure/table captions.

Citations.

File: _redirects [hunk 8]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/ioslides_presentation_format.html  https://bookdown.org/yihui/rmarkdown/ioslides-presentation.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

Figure/table captions.

File: _redirects [hunk 9]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/slidy_presentation_format.html https://bookdown.org/yihui/rmarkdown/slidy-presentation.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

Figure/table captions.

File: _redirects [hunk 10]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/beamer_presentation_format.html  https://bookdown.org/yihui/rmarkdown/beamer-presentation.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

Figure/table captions.

File: _redirects [hunk 11]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/revealjs_presentation_format.html   https://bookdown.org/yihui/rmarkdown/revealjs.html   301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

Figure/table captions.

Citations.

File: _redirects [hunk 12]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/rmarkdown_websites.html   https://bookdown.org/yihui/rmarkdown/rmarkdown-site.html  301!
```
Comment (yihui) [inline _redirects]: Levels of section headers.

Figure/table captions.

File: _redirects [hunk 13]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/rmarkdown_site_generators.html  https://bookdown.org/yihui/rmarkdown/rmarkdown-site.html  301!
```
Comment (yihui) [inline _redirects]: Merged into `/rmarkdown_websites.html`.

File: _redirects [hunk 14]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/tufte_handout_format.html   https://bookdown.org/yihui/rmarkdown/tufte-handouts.html  301!
```
Comment (yihui) [inline _redirects]: Largely rewritten, with figure/table captions and citations.

File: _redirects [hunk 15]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/package_vignette_format.html  https://bookdown.org/yihui/rmarkdown/r-package-vignette.html  301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

File: _redirects [hunk 16]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_pandoc_markdown.html    https://pandoc.org/MANUAL.html#pandocs-markdown   301!
+/authoring_rcodechunks.html   https://bookdown.org/yihui/rmarkdown/r-code.html  301!
```
Comment (yihui) [inline _redirects]: Largely rewritten, with figure captions and citations.

File: _redirects [hunk 17]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_knitr_engines.html  https://bookdown.org/yihui/rmarkdown/language-engines.html   301!
```
Comment (yihui) [inline _redirects]: Largely rewritten. Changed levels of section headers. Added citations.

File: _redirects [hunk 18]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_shiny.html  https://bookdown.org/yihui/rmarkdown/shiny-documents.html  301!
```
Comment (yihui) [inline _redirects]: Figure captions.

Reorganized content in the book.

File: _redirects [hunk 19]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_embedded_shiny.html   https://bookdown.org/yihui/rmarkdown/shiny-embedded.html  301!
```
Comment (yihui) [inline _redirects]: Merged to `/authoring_shiny.html`.

File: _redirects [hunk 20]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_shiny_widgets.html  https://bookdown.org/yihui/rmarkdown/shiny-widgets.html   301!
```
Comment (yihui) [inline _redirects]: Merged to `/authoring_shiny.html`.

File: _redirects [hunk 21]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_shiny_advanced.html   https://bookdown.org/yihui/rmarkdown/shiny-delay.html   301!
```
Comment (yihui) [inline _redirects]: Merged to `/authoring_shiny.html`.

File: _redirects [hunk 22]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/authoring_shiny_output.html   https://bookdown.org/yihui/rmarkdown/shiny-args.html  301!
```
Comment (yihui) [inline _redirects]: Merged to `/authoring_shiny.html`.

File: _redirects [hunk 23]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/developer_html_widgets.html   https://bookdown.org/yihui/rmarkdown/html-widgets.html  301!
```
Comment (yihui) [inline _redirects]: Substantially expanded into a chapter.

File: _redirects [hunk 24]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/developer_parameterized_reports.html  https://bookdown.org/yihui/rmarkdown/parameterized-reports.html  301!
```
Comment (yihui) [inline _redirects]: Figure captions.

File: _redirects [hunk 25]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/developer_document_templates.html   https://bookdown.org/yihui/rmarkdown/document-templates.html   301!
```
Comment (yihui) [inline _redirects]: Figure captions. Citations.

File: _redirects [hunk 26]
```diff
@@ -2,3 +2,31 @@ http://rmarkdown.rstudio.com/*   https://rmarkdown.rstudio.com/:splat  301!
...
+/developer_custom_formats.html   https://bookdown.org/yihui/rmarkdown/new-formats.html   301!
```
Comment (yihui) [inline _redirects]: Cross-references of sections.

**Comment by yihui** [comment]
@jjallaire What do you think?

Note I haven't rebuilt all Rmd's yet (there are a few Pandoc 2.2 issues I need to resolve first).

**Comment by jjallaire** [comment]
I'm still a bit worried that this will provide a pretty disjointed experience (but at the same time I understand we don't want to duplicate a bunch of work). Perhaps we can discuss at work week?

**Comment by yihui** [comment]
Sorry, I'm not going to the work week this time.

**Comment by jjallaire** [comment]
I think we should have a discussion with the education folks (Carl, Garrett, etc.) about how they want to handle user navigation to all of these resources. I think your chapters are all obviously better than what is on the website, I just want to have a navigational structure that isn't too disjointed.

**Comment by yihui** [comment]
Okay, I'll reach out to them.

**Comment by yihui** [comment]
So I have talked to @garrettgman and @cdhowe. Basically they are fine with opening these links in new windows. Garrett made an analogy of this website to Wikipedia: the rmarkdown website could be a portal of resources (as links) rather than the core documentation of the rmarkdown package -- if readers want to know more, they can discover the book and jump there. Of course, the analogy does not address the reading experience (all Wikipedia pages are of the same style).

**Comment by jjallaire** [comment]
Okay, then let's go ahead and make this change.

**Pull request merged**